src/Symfony/Component/ExpressionLanguage/SyntaxError.php

```diff
@@ -13,14 +13,29 @@
 
 class SyntaxError extends \LogicException
 {
-    public function __construct($message, $cursor = 0, $expression = '')
+    public function __construct($message, $cursor = 0, $expression = '', $subject = null, array $proposals = null)
     {
         $message = sprintf('%s around position %d', $message, $cursor);
         if ($expression) {
             $message = sprintf('%s for expression `%s`', $message, $expression);
         }
         $message .= '.';
 
+        if (null !== $subject && null !== $proposals) {
+            $minScore = INF;
+            foreach ($proposals as $proposal) {
+                $distance = levenshtein($subject, $proposal);
+                if ($distance < $minScore) {
+                    $guess = $proposal;
+                    $minScore = $distance;
+                }
+            }
+
+            if (isset($guess) && $minScore < 3) {
+                $message .= sprintf("\n\nDid you mean \"%s\"?", $guess);
```

Comment:
I would add a dot to end the message and remove the `\n` (that's formatting, but we have no formatting context here)

Reply:
@fmata Can you make this change? Thanks.

Reply:
@fabpot done